exposed-tests/src/test/kotlin/org/jetbrains/exposed/sql/tests/shared/dml/UpsertTests.kt

```diff
@@ -749,6 +749,24 @@ class UpsertTests : DatabaseTestsBase() {
         }
     }
 
+    @Test
+    fun testUpsertWhenColumnNameIncludesTableName() {
+        val tester = object : Table("my_table") {
```

Comment:
Minor: looks like, if the table created with `IntIdTable` the error on the master is the same. It confuses a bit, at the first time I thought that the problem is in the custom `id` column, but looks like it's not.

Reply:
Yup, it fails if `IntIdTable` is used instead, also with an invalid identifier exception, but not quite the same.

It fails because `S.ID`, despite being correct, is not declared anywhere in:
`USING (VALUES (1, 'Hello')) S(MY_TABLE_ID, MY_TABLE_VALUE)`.
The type of single-table `MERGE` used for upserts relies on a derived source table above, so key columns used in the `ON` clause need to be defined first in this table, which doesn't happen if auto-incrementing (because no value is set).
So the test passes if a manual value is also assigned to the `id` column in `upsert()`:
`USING (VALUES (1, 1, 'Hello')) S(ID, MY_TABLE_ID, MY_TABLE_VALUE)`.

It's a limitation I noticed when upsert via merge was [first implemented](https://github.com/JetBrains/Exposed/blob/main/exposed-tests/src/test/kotlin/org/jetbrains/exposed/sql/tests/shared/dml/UpsertTests.kt#L31) and I kept wondering if a user would ever come forward with it as an issue, but so far nothing.